### PR TITLE
PLANET-7563 Apply link styles to rest of body links

### PR DESCRIPTION
**Pull request by mleray**
### Description

See [PLANET-7563](https://jira.greenpeace.org/browse/PLANET-7563)

**Requirements:**

- [x] Apply the new links style to linked text in lists
- [x] Apply the new links style to linked text in tables
- [x] Apply the new links style to linked text in image captions
- [x] Apply the new links style to linked text in headings, excluding headings within blocks (articles, covers, ...)
- [x] Apply the new links style to linked text in embed blocks captions

### Testing

I've grouped all the new link cases on [this page](https://www-dev.greenpeace.org/test-iocaste/all-the-new-link-cases/), you should test it both in the editor and in the frontend. Please also check out other Pages/Posts on the instance, to make sure that nothing else is affected (especially headings in our blocks).

### CODE REVIEW

**Comment by planet-4** [comment]
### Test instance is ready :rocket:

:new_moon: [iocaste](https://www-dev.greenpeace.org/test-iocaste) | [admin](https://www-dev.greenpeace.org/test-iocaste/wp-admin/) | [blocks report](https://www-dev.greenpeace.org/test-iocaste/wp-admin/?page=plugin_blocks_report) | [CircleCI](https://app.circleci.com/pipelines/github/greenpeace/planet4-test-iocaste) | [composer-local.json](https://github.com/greenpeace/planet4-test-iocaste/blob/main/composer-local.json)

:watch: 2024.09.12 12:56:45

**Review by mardelnet — COMMENTED** [review]
Nice job, @mleray 
I have a couple of questions:

1. According to [this Figma file](https://www.figma.com/design/9NtbY8n3at8uOEJTsLrETb/P4-Design-System?node-id=370-36185&node-type=canvas&t=eO3YLMzxnaaQvrth-0), headings h3, h4, and h5 should not use the new styles. Am I looking at the right mockup?
2. The styles are not visible in the editor of List blocks.
3. The styles are not visible in image captions.
4. The styles are not visible in embed blocks captions.
5. This is very picky-nicky, but the focus state is missing the border radius.

**Comment by mleray** [comment]
> 1. According to [this Figma file](https://www.figma.com/design/9NtbY8n3at8uOEJTsLrETb/P4-Design-System?node-id=370-36185&node-type=canvas&t=eO3YLMzxnaaQvrth-0), headings h3, h4, and h5 should not use the new styles. Am I looking at the right mockup?

The mockup you are referring to with the headings is for the [Sitemap page](https://www-dev.greenpeace.org/test-iocaste/sitemap/), as far as I know that one should remain unchanged but the rest of headings should get the new link styles. I'll still ask Houssam just to be sure 🙂 He'll probably need to update these mockups anyway, with the new possible links!

File: assets/src/scss/blocks/core-overrides/Table.scss [hunk 1]
```diff
@@ -49,6 +53,7 @@
 
   // Table caption
   figcaption {
+    min-height: 24px; // This is needed in case there's a link in the caption, for our styles
```
Comment (mardelnet) [inline assets/src/scss/blocks/core-overrides/Table.scss:56]: Should we use `rem` or a variable for this value?

**Review by mardelnet — APPROVED** [review]
LGTM, @mleray 
I have only one more question (but I'm approving this anyways):

File: assets/src/scss/blocks/core-overrides/Table.scss (hunk 1 above)
Comment (mleray) [reply assets/src/scss/blocks/core-overrides/Table.scss:56]: I think it's fine, we usually don't really use those for heights 🤔 Mostly paddings, margins, and font sizes

**Pull request merged**